### ISSUE TITLE
aws cli to auto guess content type of .br (brotli) files

### ISSUE THREAD

**Issue by kazazor** [aws/aws-cli#3817]
Today AWS cli is trying to guess the content type of the file even if it's compressed (which is a VERY common case when uploading to S3 files instead of cloudfront auto compression. This is being done here: https://github.com/aws/aws-cli/blob/5d349108c974d80b5b1a634e8a7433556a31e605/awscli/customizations/s3/utils.py#L288-L307

Which internally uses a python library with hardcoded options: https://github.com/python/cpython/blob/cedc9b74202d8c1ae39bca261cbb45d42ed54d45/Lib/mimetypes.py#L402-L530

All of that is described here in a closed issue: https://github.com/aws/aws-cli/issues/3303

However, `.br` extensions are becoming more and more popular and are being sent `by default` by all major browsers (excluding IE 11). Because of that problem `.br` files content-type are not being guessed at all and we're required to do it manually.

I suggest either adding it to the python library or embedding the hard-coded list to aws-cli so you will have full control.

**Comment by JordonPhillips** [comment]
I'm -1 on maintaining a list in the CLI. The mimetypes module allows extension via an array of common files, so if you want something that isn't built in then you should edit one of those files or otherwise provide the value on the command line as you're uploading the files.

**Comment by kazazor** [comment]
I can understand why not to embed it here, even that you're not controlling your own fate on this one..

`or otherwise provide the value on the command line as you're uploading the files.`
I believe the auto detect content type was entered there for a reason. Currently, the cli does not support it anymore for one of the popular compressions out there which is the default by all major browsers.. I do believe something should be done to support it.. I have already done it manually for our repo but I do believe it's needed from the same reason it was added there to begin with.

@JordonPhillips - what other options do you see here? Just updating the python repo with this code? Seems like it will never happen to be honest :/

**Comment by JordonPhillips** [comment]
Yeah, your options are to either get it added into python or add it to your own configuration files. Have you tried opening a PR on cpython? It looks like it would be a pretty simple change.

**Comment by iAnomaly** [comment]
For anyone else looking for this, support for Brotli (.br) extension was added to the upstream Python library by [this PR](https://github.com/python/cpython/pull/12200). Sadly, unless they backport to Python 3.8 we will have to wait for 3.9 to use it.
